program/src/processor.rs

```diff
@@ -579,8 +611,42 @@ pub fn withdraw_excess_inactive_sol<'a, 'b>(
     Ok(excess_balance)
 }
 
-/// Recover any inactive balance from the validator's stake accounts to the
-/// reserve account.
+pub fn compare_stake_account<'a, 'b>(
```

Comment:
I am confused by this function signature; how can an `AccountInfo` be the outcome of a comparison? I would expect a `compare` function to return [`Ordering`](https://doc.rust-lang.org/std/cmp/enum.Ordering.html).